[arm] add test_cv demo

chenjiaoAngel opened the pull request:
add test_cv demo in demo/cxx
support read image processing and  image preprocessing when running model
support the performance test with cv image-preprocess and opencl image-preprocess

chenjiaoAngel committed:
Merge pull request #8 from PaddlePaddle/develop

update

chenjiaoAngel committed:
add cv image process

chenjiaoAngel committed:
fix arm liunx build error

chenjiaoAngel committed:
add LITE_WITH_CV defien to make cv, test=develop

chenjiaoAngel committed:
fix cv format, annd add describe in utils/cv

chenjiaoAngel committed:
delete some Meaningless comments, test=develop

chenjiaoAngel committed:
set LITE_WITH_CV=OFF in build.sh, test=develop

chenjiaoAngel committed:
delete cv_enum.h in utils/cv, push the contents in cv_ennum.h to paddle_image_preprocess.h, test=develop

chenjiaoAngel committed:
according to reviews to redefine paddle_image_preprocess.h, test=develop

chenjiaoAngel committed:
add detailed note of flipParam, test=develop

chenjiaoAngel committed:
fix format in paddle_image_preprocess.h, test=develop

chenjiaoAngel committed:
fix error when build x86. test=develop

lite_with_X86 does not contain lite_with_cv

chenjiaoAngel committed:
fix cmake error in llite/CMakeLists.txt, missing mkdir cxx, test=develop

chenjiaoAngel committed:
Merge pull request #15 from PaddlePaddle/develop

update code

chenjiaoAngel committed:
according to review change, test=develop

chenjiaoAngel committed:
chang grb to rgb, test=develop

chenjiaoAngel committed:
Merge pull request #31 from PaddlePaddle/develop

update code

chenjiaoAngel committed:
add elemetnwise mul constant elimination and deconv+relu, deconv+batchnorm fusion, test=develop

chenjiaoAngel committed:
Merge branch 'develop' into image

chenjiaoAngel committed:
fix format, test=develop

chenjiaoAngel committed:
fix model_optimize bug, update concat and split op, speed up, test=develop

chenjiaoAngel committed:
update split speed, test=develop

chenjiaoAngel committed:
fic confilct
Merge branch 'image' of https://github.com/chenjiaoAngel/Paddle-Lite into image

chenjiaoAngel committed:
merge branch
Merge branch 'image_dev' into image

chenjiaoAngel committed:
fix format, test=develop

chenjiaoAngel committed:
Merge branch 'develop' into image

chenjiaoAngel committed:
Merge branch 'develop' into image

chenjiaoAngel committed:
add classify demo inn demo/cxx/ , test=develop

chenjiaoAngel committed:
fix formart inn mobile_classify, test=develop

chenjiaoAngel committed:
git pull code and update
Merge branch 'image' of https://github.com/chenjiaoAngel/Paddle-Lite into image

chenjiaoAngel committed:
delete some note and extra code, test=develop

chenjiaoAngel committed:
Merge branch 'develop' into image

chenjiaoAngel committed:
remove test.jpg and labels.txt, test=develop

chenjiaoAngel committed:
pull code and update, test=develop
Merge branch 'image' of https://github.com/chenjiaoAngel/Paddle-Lite into image

chenjiaoAngel committed:
add test_cv in cxx/demo

chenjiaoAngel committed:
add test_cv READMEE, test=develoop

chenjiaoAngel committed:
fix coonflict, test=deveelop
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle-Lite into PaddlePaddle-develop

chenjiaoAngel committed:
Merge branch 'PaddlePaddle-developt push origin new_im
' into new_im

chenjiaoAngel committed:
add note info, flip only support x, y, xy;rotate only support 90, 180, 270; test=develop

chenjiaoAngel committed:
fix build error, paddle_cv_arm , test=develop

chenjiaoAngel committed:
add GRAY to RGBA(BGRA) convert and RGBA(BGRA)_to_Tensor, test=develop

chenjiaoAngel committed:
update code
Merge branch 'new_im' of https://github.com/chenjiaoAngel/Paddle-Lite into new_im

MyPandaShaoxiang commented on lite/demo/cxx/test_cv/test_img_prepross.cc:
```diff
@@ -0,0 +1,392 @@
+
+      // read img and pre-process
+      float means[3] = {0.485f, 0.456f, 0.406f};
+      float scales[3] = {0.229f, 0.224f, 0.225f};
```
mean和scale上面不是已经定义了？

chenjiaoAngel replied:
这两个不一样。open cv处理用的mean和scale与CV里面不一样。
opencv先是归一化，均除255， scale传进去是做除法。我的scale进去是做乘法

yiicy commented on lite/demo/cxx/README.md:
```diff
@@ -66,26 +66,42 @@ adb -s emulator-5554 pull /data/local/tmp/test_detection_result.jpg ./
+adb -s emulator-5554 push ../../../cxx/lib/libpaddle_full_api_shared.so /data/local/tmp/
+adb -s emulator-5554 shell chmod +x /data/local/tmp/mobile_classify
+adb -s emulator-5554 shell "export LD_LIBRARY_PATH=/data/local/tmp/:$LD_LIBRARY_PATH && 
+/data/local/tmp/mobile_classify /data/local/tmp/mobilenetv1opt2 /data/local/tmp/test.jpg /data/local/tmp/labels.txt"
```
这里应该是test_model? 另外加一下图像预处理库demo的运行结果吧

chenjiaoAngel replied:
这个预处理结果不用加吧，跟前面的demo写法保持一样。
另外，如果看效果话，在test_cv/README.md有写怎么查看效果

chenjiaoAngel replied:
这个预处理结果不用加吧，跟前面的demo写法保持一样。
另外，如果看效果话，在test_cv/README.md有写怎么查看效果

yiicy replied:
简单加一句吧，这个readme中之前的demo都会提一下结果

chenjiaoAngel replied:
已修改，请查阅

yiicy commented on lite/demo/cxx/README.md:
```diff
@@ -66,26 +66,42 @@ adb -s emulator-5554 pull /data/local/tmp/test_detection_result.jpg ./
+tar zxvf mobilenet_v1.tar.gz
+./model_optimize_tool optimize model
+make
+adb -s emulator-5554 push test_model /data/local/tmp/
```
test_model改成一个更相关的名字吧

chenjiaoAngel replied:
已修改，请查阅

chenjiaoAngel committed:
fix format from review, test=develop

chenjiaoAngel committed:
fix makefile format. test=devellop

chenjiaoAngel committed:
Merge branch 'develop' into new_im

chenjiaoAngel committed:
fix bbuuild v7 error, test=develop

chenjiaoAngel committed:
update coode
Merge branch 'new_im' of https://github.com/chenjiaoAngel/Paddle-Lite into new_im

MyPandaShaoxiang reviewed (dismissed):
lgtm

chenjiaoAngel committed:
Merge branch 'develop' into new_im

yiicy reviewed (approved):
LGTM
